scripts.d/ta/280_check_only_one_weka_version.sh

```diff
@@ -8,15 +8,41 @@ SCRIPT_TYPE="parallel"
 
 RETURN_CODE=0
 
+# Check if we can run weka commands
+weka status &> /dev/null
+if [[ $? -ne 0 ]]; then
+    echo "ERROR: Not able to run Weka commands."
+    exit 254
+elif [[ $? -eq 127 ]]; then
+    echo "Weka not found."
+    exit 254
+elif [[ $? -eq 41 ]]; then
+    echo "Unable to login into Weka cluster."
+    exit 254
+fi
+
+
 NUMBER_OF_WEKA_VERSIONS=$(weka version | wc -l)
 if [[ ${NUMBER_OF_WEKA_VERSIONS} -ne 1 ]] ; then
-    echo "There is more than one Weka version installed - this is usually a remnant"
-    echo "of previous upgrades and not removing older versions".
-    echo "The non-default versions can be removed if required with: "
-    for NON_DEFAULT_WEKA_VERSION in $(weka version | grep -v "^*") ; do
-        echo "    weka version rm ${NON_DEFAULT_WEKA_VERSION}"
-    done
-    RETURN_CODE=254
+
+    CONTAINER_LIST_FILE="/tmp/weka_cluster_container_versions.txt.${RANDOM}"
+    # Get the current list of container versions, in case we have older clients
+    weka cluster container --output release --no-header | sort | uniq > ${CONTAINER_LIST_FILE} 2>/dev/null
+
+    NUMBER_OF_POSSIBLY_REDUNDANT_WEKA_VERSIONS=$(weka version | grep -v -f ${CONTAINER_LIST_FILE} | wc -l)
+    if [[ ${NUMBER_OF_POSSIBLY_REDUNDANT_WEKA_VERSIONS} -ne 1 ]] ; then
+        echo "There is more than one Weka version installed - this is usually a remnant"
+        echo "of previous upgrades and not removing older versions".
+        echo "The non-default versions can be removed if required with: "
```

Comment:
Possibly emphasize this is a "local" command and must be ran on each backend?

Reply:
I have changed the wording.